### ISSUE TITLE
When there is a network issue, 'No Bookings Found' is displayed, instead of an error

### ISSUE THREAD

**Issue by kantord** [kiwicom/smart-faq#293]
When booking api is down, there should be an error displayed, but this is shown instead:
![2018-05-31-151909_2205x950_scrot](https://user-images.githubusercontent.com/3704904/40784580-51d3e786-64e6-11e8-95ee-5805cfb3de51.png)

This might be confusing to users and might be generating extra customer service traffic. We should display something like "Network error. Please try again in a few minutes. For urgent inquiries don't hesitate to call customer service at {phone number}"

**Comment by jalbertsr** [comment]
Notice, that there is already a `BookingError` designed (and implemented) component for this, the issue should be investigating why was returning `BookingNotFound` instead of `BookingError`

**Comment by edgarordonez** [comment]
Hi @kantord, @jalbertsr 

I can't reproduce this error, returns a BookingError component. What do we do?

Thanks!

**Comment by kantord** [comment]
Unfortunately I cannot describe the error much better, as I only know it happened during (and probably because) of the API downtime. The error might actually be in the graphql repo though. 

```javascript
renderBooking = (renderState: RenderState) => {
    if (renderState && renderState.error) {
      return <BookingError />;
    }

    if (!renderState.props) {
      return <Loader />;
    }

    const booking = renderState.props.nearestBooking;

    if (!booking) {
      return <BookingNotFound />;
    }

    return <BookingDetail booking={booking} />;
  };
```

Based on this code, renderState should have an error when the booking api is down. I think there's a chance that the graphql server wasn't properly handling the downtime and didn't produce the error. It might worth a shot checking out how the graphql server behaves on this query when it cannot access the booking api (change the booking api URL to something made up maybe)? Let's see if that helps with reproducing the issue.

**Comment by kantord** [comment]
@jaroslav-kubicek, because of the difficulty of reproducing this issue and the rarity of the error that causes it, could this maybe done in the next sprint? I mean probably we can release without this, since if booking api goes down in the first week of production, this might be our smallest worry :-)

**Comment by mrtnzlml** [comment]
@edgarordonez showed me that it returns this in case of failure:

```json
{
  "data": {
    "nearestBooking": null
  }
}
```

I am sorry to disappoint you all but in this case, Relay does not return `error` field (it's always `null`). Relay uses this field only in case of total server failure which is basically this:

```json
{
  "data": null
}
```

So you cannot really rely on `renderState.error` here.

**Comment by kantord** [comment]
Well, at least we can reproduce the issue now.

I'm not sure what the most idiomatic solution would be, but I think we can either detect the error on the graphql server, so maybe we can create a special value when there's an error?

**Comment by mrtnzlml** [comment]
@kantord What are you trying to solve here? Can you please elaborate the situation more? Field `nearestBooking` is `null` - render your UI according to this situation. You don't need to work with the error here, do you?

**Comment by kantord** [comment]
You're right, we don't need to work with the error or know much about it besides that it exists. We just need to display the error message *if* we cannot react the booking data through GraphQL. It already happens if we cannot reach GraphQL itself, if my understanding is correct. This bug happened when GraphQL server was working fine but the booking API was down. If `nearestBooking === null` means there was an error, then problem solved, just change the conditions and done. But is that situation the only reason for `nearestBooking` to be `null`? If not, we need more information for our condition to be able to deduce the situation and correctly handle it.

**Comment by mrtnzlml** [comment]
Well, that really depends on your needs. If you want to work with the partial error in your component you'll have to patch the Relay itself (related: https://github.com/facebook/relay/issues/1913). We actually recently removed this feature from `kiwicom/mobile` because it was not needed (yet).

**Comment by MagdalenaAlmanova** [comment]
I think, that it is ok.

**Comment by jaroslav-kubicek** [comment]
If I'm correct, this wasn't actually fixed, see #616

**Comment by kantord** [comment]
I agree

**Comment by jaroslav-kubicek** [comment]
damn automation 🤦‍♂️
